nomic/aws/sagemaker.py

```diff
@@ -180,3 +185,139 @@ def embed_texts(
         "model": "nomic-embed-text-v1.5",
         "usage": {},
     }
+
+
+def preprocess_images(images: List[Union[str, "PIL.Image.Image", bytes]]) -> List[bytes]:
+    """
+    Preprocess a list of images for embedding using a sagemaker model.
+
+    Args:
+        images: List of images to be embedded.
+
+    Returns:
+        List of images formatted for sagemaker embedding.
+    """
+    encoded_images = []
+    for image in images:
+        if isinstance(image, str):
+            image = PIL.Image.open(image)
+        elif isinstance(image, bytes):
+            image = PIL.Image.open(io.BytesIO(image))
+        elif isinstance(image, PosixPath):
+            image = PIL.Image.open(image)
+        else:
+            assert isinstance(image, PIL.Image.Image), f"Invalid image type: {type(image)}"
+        image = image.convert("RGB")
+        buffered = io.BytesIO()
+        image.save(buffered, format="JPEG")
+        encoded_image = buffered.getvalue()
+        encoded_images.append(encoded_image)
+    return encoded_images
+
+
+def sagemaker_image_request(image: Union[str, bytes, "PIL.Image.Image"], sagemaker_endpoint: str, region_name: str):
+    preprocessed_image = preprocess_images([image])
+
+    client = boto3.client("sagemaker-runtime", region_name=region_name)
+    response = client.invoke_endpoint(
+        EndpointName=sagemaker_endpoint, Body=preprocessed_image[0], ContentType="image/jpeg"
+    )
+
+    return parse_sagemaker_response(response)
+
+
+def embed_image(
```

Comment:
nit: `embed_images` instead?

Reply:
yeah it's a little odd our python client has `embed.text` as well as `embed.images` while in sagemaker we have `embed_texts` and `embed_image`. I'll fix to mirror what our api has, thanks for pointing this out!

Reply:
we should change our python client signature to embed.image

Reply:
why?

Reply:
updated